app/gui/src/presenter.rs

```diff
@@ -125,11 +138,37 @@ impl Model {
                 } else {
                     warn!("Project creation failed: no ProjectManagingAPI available.");
                 }
-            })
+            }))
         } else if let Some(template) = template {
             error!("Invalid project template name: {template}");
         };
     }
+
+    /// Open a project by name or ID. If no project with the given name exists, it will be created.
+    #[profile(Task)]
+    fn open_or_create_project(&self, project: ProjectToOpen) {
+        let controller = self.controller.clone_ref();
+        crate::executor::global::spawn(with_progress_indicator(|| async move {
+            if let Ok(managing_api) = controller.manage_projects() {
+                if let Err(error) = managing_api.open_or_create_project(project).await {
+                    error!("Cannot open or create project: {error}.");
+                }
+            } else {
+                warn!("Project opening failed: no ProjectManagingAPI available.");
+            }
+        }));
+    }
+}
+
+/// Show a full-screen spinner for the exact duration of the specified function.
+async fn with_progress_indicator<F, T>(f: F)
+where
+    F: FnOnce() -> T,
+    T: Future<Output = ()>, {
+    let Ok(app) = js::app() else { return error!("Failed to get JavaScript EnsoGL app.") };
+    app.show_progress_indicator(OPEN_PROJECT_SPINNER_PROGRESS);
+    f().await;
+    app.hide_progress_indicator();
```

Comment:
1. We had an old code that was doing `js::app()` but I can't see it changed in this PR. What happened with it? 
2. I believe that we are doing `js::app()` somewhere in EnsoGL core already. Doing it twice probably does not make sense as it is more error prone. We need JS app there, so the app should already be initialized.

Reply:
> 1. We had an old code that was doing `js::app()` but I can't see it changed in this PR. What happened with it?
> 2. I believe that we are doing `js::app()` somewhere in EnsoGL core already. Doing it twice probably does not make sense as it is more error prone. We need JS app there, so the app should already be initialized.

`js::app` is a getter, isn't it? Not an initializer. If it's initialized, then I don't know how could we access it.

Reply:
Right. It's a getter, but it is kind of an unsafe getter - it gets a variable from JS, casts it to a type, etc. Somewhere in EnsoGL we just have already a logic of getting app and throwing an error if it's not defined. This is a logic duplication. It would be good not to duplicate it.

Reply:
Let's mark it as "TODO" right now to refactor properly, because we must move to more pressing issues.